tools/azsdk-cli/Azure.Sdk.Tools.Cli/Services/Languages/DotNetLanguageSpecificChecks.cs

```diff
@@ -50,28 +45,15 @@ public async Task<CLICheckResponse> CheckGeneratedCode(string packagePath, bool
                 return new CLICheckResponse(1, "", "Failed to determine service directory from the provided package path.");
             }
 
-            var repoRoot = _gitHelper.DiscoverRepoRoot(packagePath);
-            var scriptPath = Path.Combine(repoRoot, "eng", "scripts", "CodeChecks.ps1");
-            if (!File.Exists(scriptPath))
-            {
-                _logger.LogError("Code checks script not found at: {ScriptPath}", scriptPath);
-                return new CLICheckResponse(1, "", $"Code checks script not found at: {scriptPath}");
-            }
-
-            var args = new[] { scriptPath, "-ServiceDirectory", serviceDirectory, "-SpellCheckPublicApiSurface" };
-            var options = new PowershellOptions(scriptPath, args, workingDirectory: repoRoot, timeout: CodeChecksTimeout);
-            var result = await _powershellHelper.Run(options, ct);
+            var (invoked, result) = await _repositoryScriptService.TryInvoke("CodeChecks", packagePath, new()
```

Comment:
I was not expecting this to be a script invoke wrapper that is called by language specific tools. I was expecting this to be handled in the language service itself. For example with something like UpdateMetadata tool we would look for a script that handles the in the config and would expect a specific set of parameters (i.e. repo root and package path) and the language would implement a script or update an existing script to take those parameters to do the work.